scripts/api-test-comprehensive.sh

```
@@ -0,0 +1,254 @@
+    
+    if [ ! -z "$headers" ]; then
+        curl_cmd="$curl_cmd $headers"
+    fi
+    
+    if [ ! -z "$data" ]; then
+        curl_cmd="$curl_cmd -d '$data'"
+    fi
+    
+    curl_cmd="$curl_cmd $BASE_URL$endpoint"
+    
+    local response=$(eval $curl_cmd)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)